File: src/Http.API/ClientApp/src/app/share/services/entity-member-constraint.service.ts
import { Injectable } from '@angular/core';
import { BaseService } from './base.service';
import { Observable } from 'rxjs';
import { EntityMemberConstraintFilterDto } from '../models/entity-member-constraint/entity-member-constraint-filter-dto.model';
import { EntityMemberConstraintAddDto } from '../models/entity-member-constraint/entity-member-constraint-add-dto.model';
import { EntityMemberConstraintUpdateDto } from '../models/entity-member-constraint/entity-member-constraint-update-dto.model';
import { EntityMemberConstraintItemDtoPageList } from '../models/entity-member-constraint/entity-member-constraint-item-dto-page-list.model';
import { EntityMemberConstraint } from '../models/entity-member-constraint/entity-member-constraint.model';

/**
 * 属性的约束
 */
@Injectable({ providedIn: 'root' })
export class EntityMemberConstraintService extends BaseService {
  /**
   * 筛选
   * @param data EntityMemberConstraintFilterDto
   */
  filter(data: EntityMemberConstraintFilterDto): Observable<EntityMemberConstraintItemDtoPageList> {
    const url = `/api/EntityMemberConstraint/filter`;
    return this.request<EntityMemberConstraintItemDtoPageList>('post', url, data);
  }

  /**
   * 新增
   * @param data EntityMemberConstraintAddDto
   */
  add(data: EntityMemberConstraintAddDto): Observable<EntityMemberConstraint> {
    const url = `/api/EntityMemberConstraint`;
    return this.request<EntityMemberConstraint>('post', url, data);
  }

  /**
   * 更新
   * @param id 
   * @param data EntityMemberConstraintUpdateDto
   */
  update(id: string, data: EntityMemberConstraintUpdateDto): Observable<EntityMemberConstraint> {
    const url = `/api/EntityMemberConstraint/${id}`;
    return this.request<EntityMemberConstraint>('put', url, data);
  }

  /**
   * 详情
   * @param id 
   */
  getDetail(id: string): Observable<EntityMemberConstraint> {
    const url = `/api/EntityMemberConstraint/${id}`;
    return this.request<EntityMemberConstraint>('get', url);
  }

  /**
   * ⚠删除
   * @param id 
   */
  delete(id: string): Observable<EntityMemberConstraint> {
    const url = `/api/EntityMemberConstraint/${id}`;
    return this.request<EntityMemberConstraint>('delete', url);
  }

}

File: src/Sdks/Dusi.Manage.Client/Services/EntityMemberConstraintService.cs
using Core.Entities.EntityDesign;
using Share.Models.EntityMemberConstraintDtos;
namespace Dusi.Manage.Client.Services;
/// <summary>
/// 属性的约束
/// </summary>
public class EntityMemberConstraintService : BaseService
{
    public EntityMemberConstraintService(HttpClient httpClient) : base(httpClient)
    {

    }
    /// <summary>
    /// 筛选
    /// </summary>
    /// <param name="data">EntityMemberConstraintFilterDto</param>
    /// <returns></returns>
    public async Task<PageList<EntityMemberConstraintItemDto>?> Filter(EntityMemberConstraintFilterDto data) {
        var url = $"/api/admin/EntityMemberConstraint/filter";
        return await PostJsonAsync<PageList<EntityMemberConstraintItemDto>?>(url, data);
    }

    /// <summary>
    /// 新增
    /// </summary>
    /// <param name="data">EntityMemberConstraintAddDto</param>
    /// <returns></returns>
    public async Task<EntityMemberConstraint?> Add(EntityMemberConstraintAddDto data) {
        var url = $"/api/admin/EntityMemberConstraint";
        return await PostJsonAsync<EntityMemberConstraint?>(url, data);
    }

    /// <summary>
    /// 更新
    /// </summary>
    /// <param name="id"> </param>
    /// <param name="data">EntityMemberConstraintUpdateDto</param>
    /// <returns></returns>
    public async Task<EntityMemberConstraint?> Update(string id, EntityMemberConstraintUpdateDto data) {
        var url = $"/api/admin/EntityMemberConstraint/{id}";
        return await PutJsonAsync<EntityMemberConstraint?>(url, data);
    }

    /// <summary>
    /// 详情
    /// </summary>
    /// <param name="id"> </param>
    /// <returns></returns>
    public async Task<EntityMemberConstraint?> GetDetail(string id) {
        var url = $"/api/admin/EntityMemberConstraint/{id}";
        return await GetJsonAsync<EntityMemberConstraint?>(url);
    }

    /// <summary>
    /// ⚠删除
    /// </summary>
    /// <param name="id"> </param>
    /// <returns></returns>
    public async Task<EntityMemberConstraint?> Delete(string id) {
        var url = $"/api/admin/EntityMemberConstraint/{id}";
        return await DeleteJsonAsync<EntityMemberConstraint?>(url);
    }

}